src/python/pants/jvm/resolve/common.py

```diff
@@ -149,6 +151,7 @@ class ArtifactRequirement:
 
     url: str | None = None
     jar: JvmArtifactJarSourceField | None = None
+    excludes: frozenset[str] | None = None
```

Comment:
Other Pants code usually uses `FrozenOrderedSet` in dataclasses and not `frozenset`. Looking at source for `FrozenOrderedSet`, it may be for backwards compatibility to Python 3.6 (which is no longer necessary since Pants is 3.7+ now). 

cc @Eric-Arellano / @stuhood for context

Reply:
I went with `frozenset` for equality checking purposes, `excludes=["a--b", "c--d"]` is the same as `excludes=["c--d", "a--b"]`.

Reply:
I'm honestly not certain anymore why `FrozenOrderedSet` was necessary, as the author of the class....At the time I was thinking that the no-deterministic ordering of `frozenset` mattered with the engine. But that is not true, only `__hash__` and `__eq__` being deterministic matter.

In fact, I think we're over-using `FrozenOrderedSet` and calling `sorted()` way too much for no reason. This slows down Pants performance a lot.

Related is that `FrozenDict`'s `__eq__` considers order, which means we must call `sorted` much more than should be necessary.